velox/exec/AggregateWindow.cpp

```diff
@@ -296,9 +334,22 @@ class AggregateWindowFunction : public exec::WindowFunction {
       auto frameEndIndex = frameEndsVector[i] - minFrame + 1;
       computeAggregate(rows, frameStartIndex, frameEndIndex);
       result->copy(aggregateResultVector_.get(), resultOffset + i, 0, 1);
-    }
+    });
+
+    // Set null values for empty (non valid) frames in the output block.
+    setEmptyFramesResults(validRows, resultOffset, result);
   }
 
+  void setEmptyFramesResults(
+      const SelectivityVector& validRows,
+      vector_size_t resultOffset,
+      const VectorPtr& result) {
+    // Rows with empty (not-valid) frames have nullptr in the result.
+    SelectivityVector invalidRows(validRows.size());
```

Comment:
Consider re-using SelectivityVector (by introducing a member variable) and skipping this logic altogether when validRows.isAllSelected()